crates/ide_db/src/search.rs

```diff
@@ -252,23 +267,33 @@ impl<'a> FindUsages<'a> {
 
     pub fn at_least_one(self) -> bool {
         let mut found = false;
-        self.search(&mut |_reference| {
+        self.search(&mut |_, _| {
             found = true;
             true
         });
         found
     }
 
-    pub fn all(self) -> Vec<Reference> {
-        let mut res = Vec::new();
-        self.search(&mut |reference| {
-            res.push(reference);
+    /// The [`FileReferences`] returned always have unique [`FileId`]s.
+    pub fn all(self) -> Vec<FileReferences> {
```

Comment:
```suggestion
    pub fn all(self) -> FxHashMap<FileId, Vec<FileReference>> {
```

that way, we guarantee no duplicates. Maybe it makes sense to add

```
pub struct SearchResults {
    map: FxHashMap<FileId, Vec<FileReference>>
}
```

Reply:
Right, not sure how using a Hashmap slipped my mind here.